libraries/AP_ESC_Telem/AP_ESC_Telem_HobbyWing.cpp

```diff
@@ -0,0 +1,165 @@
+#include "AP_ESC_Telem_config.h"
+
+#if AP_ESC_TELEM_HOBBYWING_ENABLED
+
+#include "AP_ESC_Telem_HobbyWing.h"
+#include "AP_HobbyWing_ESC_Platinum_PRO_v3.h"
+#include "AP_HobbyWing_ESC_Platinum_v4.h"
+#include "AP_HobbyWing_ESC_XRotor_v4.h"
+
+#include <AP_SerialManager/AP_SerialManager.h>
+
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AP_ESC_Telem_HobbyWing::var_info[] = {
+#if AP_HOBBYWING_PLATINUM_PRO_V3_ENABLED
+    // @Param: V3_MASK
+    // @DisplayName: HobbyWingESC v3 Channel Bitmask
+    // @Description: Mask of which channels are returning data on serial ports
+    // @Bitmask: 0:Channel1,1:Channel2,2:Channel3,3:Channel4,4:Channel5,5:Channel6,6:Channel7,7:Channel8,8:Channel9,9:Channel10,10:Channel11,11:Channel12,12:Channel13,13:Channel14,14:Channel15,15:Channel16, 16:Channel 17, 17: Channel 18, 18: Channel 19, 19: Channel 20, 20: Channel 21, 21: Channel 22, 22: Channel 23, 23: Channel 24, 24: Channel 25, 25: Channel 26, 26: Channel 27, 27: Channel 28, 28: Channel 29, 29: Channel 30, 30: Channel 31, 31: Channel 32
+    // @User: Advanced
+    // @RebootRequired: True
+    AP_GROUPINFO("V3_MASK",  2, AP_ESC_Telem_HobbyWing, channel_mask_v3, 0),
```

Comment:
Urg, I see what you mean about the parameters. What I was thinking was that we should create a "motor custom configuration" class a bit like Pete's "custom rotation". The custom configuration would have a mask, a pole number and an output type. You could then limit the number of configurations you allow. This would avoid some of the explosion of parameters and make it easier to have mixed types.

Reply:
I was thinking of adding an enable parameter at the top of this parameter thingy.

I'm a bit worried your proposed solution will need to a cross-product problem.

One option we discussed at SV today was having *basic* configurations selectable via enumeration values and requiring scripting to configure anything more signficant.

Note that the current situation only allows you to select poles-per-backend-type, so something like the Arace Griffin would need some scripting assistance.

Also note that we kind of expect most people to use a Periph to turn these crappy serial protocols into less-crappy DroneCAN messages, so this configuration won't be applicable to most users.  Notable exception being Heli users who I think like using HobbyWing ESCs and they're pretty frugal with their motor counts so don't really need a Periph.